Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacter.cs

```diff
@@ -72,6 +72,7 @@ public override void OnNetworkSpawn()
                 NetState.ReceivedClientInput += OnClientMoveRequest;
                 NetState.OnStopChargingUpServer += OnStoppedChargingUp;
                 NetState.NetworkLifeState.LifeState.OnValueChanged += OnLifeStateChanged;
+                OnLifeStateChanged(LifeState.Alive, NetState.NetworkLifeState.LifeState.Value);
```

Comment:
Calling this here directly is pretty clean, IMO. Can we guarantee that `OnValueChanged` on a `NetworkBehaviour` won't be fired before `OnNetworkSpawn()`? And if it did, can you guarantee that the `IsServer` is valid as well? (since you'd have to check that per `OnValueChanged` invocation as the callback will be fired on clients + server). I personally like gating the callback to just the server.

Reply:
On an unrelated note, however, I found an issue with my solution (happens if the hp and lifestate is initialized on the netcharstate and if it is done in serverchar). Basically, when an unconscious player reconnects, the host will notice it and set its lifestate correctly, then sends a trigger to the NetworkAnimator, but somehow the trigger is not handled on the host, only on the client.
If instead, I initialize the hp and lifestate one frame later (i.e. via a coroutine with a yield return null), it fixes it

Reply:
I pushed the fix for it, but I'm not convinced by it. We may want to investigate this further to see if this is a bug from NetworkAnimator or something similar.

Reply:
yeah that sounds like a bug the way you're describing it, I'd expect NA to always trigger on host.
wanna create a jira and resolve this conversation after?

Reply:
Was already done [below](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/583#discussion_r843194452). 😃